src/Components/Server/src/Circuits/RemoteNavigationManager.cs

```diff
@@ -109,28 +109,18 @@ async Task PerformNavigationAsync()
             }
             catch (Exception ex)
             {
-                // We shouldn't ever reach this since exceptions thrown from handlers are caught in InvokeLocationChangingHandlerAsync.
+                // We shouldn't ever reach this since exceptions thrown from handlers are handled in HandleLocationChangingHandlerException.
                 // But if some other exception gets thrown, we still want to know about it.
                 Log.NavigationFailed(_logger, uri, ex);
+                UnhandledException.Invoke(this, ex);
```

Comment:
Does this also require a `null` check?

```suggestion
                UnhandledException?.Invoke(this, ex);
```